### PR TITLE
Make typescript maker respect tsconfig.json settings

### PR DESCRIPTION
**Pull request by chrisnicola**
Applies the suggested fix from #499

### CODE REVIEW

File: autoload/neomake/makers/ft/typescript.vim [hunk 1]
```diff
@@ -7,8 +7,9 @@ endfunction
 function! neomake#makers#ft#typescript#tsc()
     return {
         \ 'args': [
-            \ '-m', 'commonjs', '--noEmit'
+            \ '--project', getcwd(), '--noEmit'
```
Comment (blueyed) [inline autoload/neomake/makers/ft/typescript.vim]: The `getcwd()` looks weird..  it should use the buffer's dir or something probably?
`getcwd()` might return e.g. a local cwd.

Do we need something to get a project's root for this?  (see e.g. https://github.com/dbakker/vim-projectroot)

Comment (blueyed) [inline autoload/neomake/makers/ft/typescript.vim]: If the dir of the file would be enough I think something like `%p:h` could be used.

Comment (chrisnicola) [inline autoload/neomake/makers/ft/typescript.vim]: No TSC will traverse up the directory tree until it finds the first tsconfig.json. I'll try what you suggested

Comment (blueyed) [inline autoload/neomake/makers/ft/typescript.vim]: See the go makers IIRC.

Comment (blueyed) [inline autoload/neomake/makers/ft/typescript.vim]: Also: https://github.com/neomake/neomake/pull/574/files#diff-5dd22f4fc6d3cb6db8065eaddd1aa749R10

**Comment by blueyed** [comment]
Shouldn't this become a project/directory checker then, when it compiles/checks the whole project?!

**Comment by blueyed** [comment]
https://github.com/Microsoft/TypeScript/issues/5858#issuecomment-255185701 recommends:

> your best bet is to specify a new tsconfig.json extends the exiting one, and add your file to it.

Would it be possible to create such a config on the fly?  (assuming that we have a function to find / configure the main `tsconfig.json`)

**Comment by blueyed** [comment]
See https://github.com/neomake/neomake/pull/731.

https://github.com/neomake/neomake/pull/494 also is about a project maker, but would be obsolete / the same.

**Comment by gsanches** [comment]
I work with a general workspace. My solution was the code below:

```viml
function! neomake#makers#ft#typescript#tsc()
      return {
                  \ 'args': ['--project', getcwd() . '/' . substitute(system('echo ' . expand('%:.') . ' | cut -d "/" -f1'), '\n\+$', '', ''), '--noEmit'],
                  \ 'append_file': 0,
                  \ 'errorformat':
                  \   '%E%f %#(%l\,%c): error %m,' .
                  \   '%E%f %#(%l\,%c): %m,' .
                  \   '%Eerror %m,' .
                  \   '%C%\s%\+%m'
                  \ }
endfunction
```

I had to trim the system output.

**Pull request closed without merging**